bigquery/tables.js

```diff
@@ -0,0 +1,145 @@
+// https://googlecloudplatform.github.io/gcloud-node/#/docs/guides/authentication
+var BigQuery = require('@google-cloud/bigquery');
+var Storage = require('@google-cloud/storage');
+var projectId = process.env.GCLOUD_PROJECT;
+
+// Instantiate the BigQuery and Storage clients
+var bigquery = BigQuery();
+var storage = Storage({
+  projectId: projectId
+});
+// [END auth]
+
+// [START export_table_to_gcs]
+/**
+ * Export a table from BigQuery to Google Cloud Storage.
+ *
+ * @param {object} options Configuration options.
+ * @param {string} options.bucket A Google Cloud Storage bucket to use for storage.
+ * @param {string} options.file The file to save results to within Google Cloud Storage.
+ * @param {string} options.dataset The ID of the dataset to use.
+ * @param {string} options.table The ID of the project to use.
+ * @param {string} options.format Format to export as - either 'CSV', 'JSON', or 'AVRO'.
+ * @param {boolean} [options.gzip] Optional. Whether or not data should be compressed using GZIP.
+ * @param {function} callback Callback function to receive query results.
+ */
+function exportTableToGCS (options, callback) {
+  var gcsFileObj = storage.bucket(options.bucket).file(options.file);
+
+  // Export table
+  // See https://googlecloudplatform.github.io/gcloud-node/#/docs/google-cloud/latest/bigquery/table?method=export
+  var table = bigquery.dataset(options.dataset).table(options.table);
+  table.export(
+    gcsFileObj,
+    { format: options.format, gzip: options.gzip },
+    function (err, job) {
+      if (err) {
+        return callback(err);
+      }
+
+      console.log('ExportTableToGCS: submitted job %s!', job.id);
+      return callback(null, job);
+    }
+  );
+}
+// [END export_table_to_gcs]
+
+// [START poll_export_job]
+/**
+ * Check the status of a BigQuery table export job.
+ *
+ * @param {string} jobId The ID of the export job to poll.
+ * @param {function} callback Callback function to execute when the table is exported.
+ */
+function pollExportJob (jobId, callback) {
+  var job = bigquery.job(jobId);
+  job.getMetadata(function (err, metadata) {
+    if (err) {
+      return callback(err);
+    }
+    console.log('PollExportJob: job status: %s', metadata.status.state);
+
+    // If job is done, return metadata; if not, return an error.
+    if (metadata.status.state === 'DONE') {
+      return callback(null, metadata);
+    } else {
+      return callback(new Error('Job %s is not done', jobId));
+    }
+  });
+}
+// [END poll_export_job]
+// [END complete]
+
+// The command-line program
+var cli = require('yargs');
+
+var program = module.exports = {
+  exportTableToGCS: exportTableToGCS,
+  pollExportJob: pollExportJob,
+  main: function (args) {
+    // Run the command-line program
+    cli.help().strict().parse(args).argv;
+  }
+};
+
+cli
+  .command('export <bucket> <file> <dataset> <table> [--format] [--gzip]', 'Export a table from BigQuery to Google Cloud Storage.', {}, function (options) {
```

Comment:
I'm not sure you want `[--format] [--gzip]` here. I think `yargs` might interpret them as optional positional arguments or as boolean options. `node tables export --help` will generate help text that describes the command's options.